subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/configurations/DefaultConfiguration.java

```diff
@@ -1528,39 +1547,54 @@ private synchronized void ensureResolved() {
             this.lenient = lenient;
         }
 
+        private <T> T withMutableProject(final Factory<T> factory) {
+            if (domainObjectContext.getProjectPath() != null) {
```

Comment:
When is this null?

Reply:
When evaluating an init script or settings script, which are single threaded anyways.

Reply:
Done.